pkg/archive/archive_test.go

```diff
@@ -23,13 +23,15 @@ import (
 	"gotest.tools/v3/skip"
 )
 
-var tmp string
-
-func init() {
-	tmp = "/tmp/"
-	if runtime.GOOS == "windows" {
-		tmp = os.Getenv("TEMP") + `\`
+// toUnix converts the given path to forward-slashes with the drive letter removed,
+// to allow using it for paths used in "sh". It is a no-op on non-Windows. The
+// given path must be absolute and on the system-drive.
+func toUnix(t *testing.T, filePath string) string {
+	unixPath, err := checkSystemDriveAndRemoveDriveLetter(filePath)
+	if err != nil {
+		t.Fatal(err)
 	}
+	return filepath.ToSlash(unixPath)
 }
```

Comment:
I forgot I had this PR, but I ended up implementing roughly the same in https://github.com/moby/go-archive/commit/0d01df4602b5662e7090ccfa411ac0b5f28aae9b, but with some additional logic for Windows (for cases where it's not using `C:\` in CI)

- https://github.com/moby/go-archive/pull/5